src/globals.d

```diff
@@ -14,26 +14,50 @@ import ddmd.root.array;
 import ddmd.root.filename;
 import ddmd.root.outbuffer;
 
-template xversion(string s)
-{
-    enum xversion = mixin(`{ version (` ~ s ~ `) return true; else return false; }`)();
-}
-
 private string stripRight(string s)
 {
     while (s.length && (s[$ - 1] == ' ' || s[$ - 1] == '\n' || s[$ - 1] == '\r'))
         s = s[0 .. $ - 1];
     return s;
 }
 
-enum IN_GCC     = xversion!`IN_GCC`;
+immutable bool TARGET_LINUX, TARGET_OSX, TARGET_FREEBSD, TARGET_OPENBSD,
+               TARGET_SOLARIS, TARGET_WINDOS, TARGET_POSIX;
 
-enum TARGET_LINUX   = xversion!`linux`;
-enum TARGET_OSX     = xversion!`OSX`;
-enum TARGET_FREEBSD = xversion!`FreeBSD`;
-enum TARGET_OPENBSD = xversion!`OpenBSD`;
-enum TARGET_SOLARIS = xversion!`Solaris`;
-enum TARGET_WINDOS  = xversion!`Windows`;
+shared static this()
```

Comment:
Cross-compilation will require runtime variables, and cannot be done with `static if`.  This is a step towards cross-compilation, which is a forward goal, but it obviously is only a step: it doesn't get us all the way there, partially because of those non-portable library modules in the frontend.